### PR TITLE
code review changes Add GetCustomErr function for centralized error handling in DataCenterClient

### PR DESCRIPTION
**Pull request by mchavez**
# Pull Request

To help us get this pull request reviewed and merged quickly, please be sure to include the following items:

* [ ] Tests (if applicable)
* [ ] Documentation (if applicable)
* [ ] Changelog entry
* [ ] A full explanation here in the PR description of the work done

## PR Type
What kind of change does this PR introduce?
code review changes
* [x] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] Tests
* [ ] Other

## Backward Compatibility

Is this change backward compatible with the most recently released version? Does it introduce changes which might change the user experience in any way? Does it alter the Connector in any way?

* [ ] Yes (backward compatible)
* [x] No (breaking changes)

#### Useful links:

- [https://github.com/ConductorOne/baton-sdk/wiki/Coding-Guidelines](Baton SDK coding guidelines)
- [https://github.com/ConductorOne/baton/blob/main/CONTRIBUTING.md](New contributor guide)

## What's new?
-code review changes

## Summary by CodeRabbit

- **New Features**
	- Introduced centralized error handling for HTTP requests, improving consistency and clarity in error reporting across the DataCenterClient.

- **Bug Fixes**
	- Enhanced the reliability of error responses, ensuring accurate error information is provided to users.

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -193,6 +193,27 @@ func New(ctx context.Context, baseUrl string, bitbucketClient *DataCenterClient)
...
+			bbErr.ErrorSummary = fmt.Sprintf("Error reading response body %s", err.Error())
+			return bbErr
+		}
+
+		bbErr.ErrorCode = resp.StatusCode
+		bbErr.ErrorSummary = string(bodyBytes)
+	}
+
+	return bbErr
+}
```
[the remark this answers is not in the document]
Comment (ggreer) [reply pkg/client/client.go:215]: resp.Body is a nopreader so we don't need to close it.

File: pkg/client/client.go [hunk 2]
```diff
@@ -193,6 +193,27 @@ func New(ctx context.Context, baseUrl string, bitbucketClient *DataCenterClient)
...
+		ErrorMessage:     err.Error(),
+		ErrorDescription: err.Error(),
+		ErrorLink:        req.URL.String(),
+	}
+
+	if resp != nil {
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			bbErr.ErrorSummary = fmt.Sprintf("Error reading response body %s", err.Error())
+			return bbErr
```
Comment (ggreer) [inline pkg/client/client.go:207]: bbErr.ErrorCode should be set before we return. I recommend setting it before line 204.

Comment (mchavez) [reply pkg/client/client.go:215]: Fixed

File: pkg/client/client.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mchavez) [reply pkg/client/client.go]: Fixed

**Review by ggreer — APPROVED** [review]

**Pull request merged**